codegen/gateway.go

```diff
@@ -427,18 +427,20 @@ func NewEndpointSpec(
 	}
 
 	dirName := filepath.Base(filepath.Dir(jsonFile))
-	goFolderName := filepath.Join(
-		h.CodeGenTargetPath(),
-		"endpoints",
-		dirName,
-	)
 
-	goStructsFileName := filepath.Join(
-		h.CodeGenTargetPath(),
-		"endpoints",
-		dirName,
-		dirName+"_structs.go",
-	)
+	goFolderName, err := h.GoFileGenPath(thriftFile, "")
```

Comment:
Why ? We don't want this change. The file name should be based on `client-config.json` and `endpoints-config.json` , not the thrift file.